rules/ruleset_json_test.go

```diff
@@ -0,0 +1,522 @@
+	mappingRules := []MappingRuleJSON{
+		*testMappingRuleJSON("", "mr"),
+		*testMappingRuleJSON("", "mr"),
+	}
+	rollupRules := []RollupRuleJSON{
+		*testRollupRuleJSON("", "rr", []RollupTargetJSON{*testRollupTargetJSON("target")}),
+		*testRollupRuleJSON("", "rr", []RollupTargetJSON{*testRollupTargetJSON("target")}),
+	}
+	fixture := testRuleSetJSON("namespace", mappingRules, rollupRules)
+
+	actual, err := fixture.RuleSetSnapshot(GenerateID)
+	require.NoError(t, err)
+	mrIDs := []string{}
+	rrIDs := []string{}
+
+	// Test that generated IDs are UUIDs and add them to their respective lists for further testing.
+	for id := range actual.MappingRules {
+		require.NotNil(t, uuid.Parse(id))
+		mrIDs = append(mrIDs, id)
+	}
+	for id := range actual.RollupRules {
+		require.NotNil(t, uuid.Parse(id))
+		rrIDs = append(rrIDs, id)
+	}
+
+	expected := &RuleSetSnapshot{
+		Namespace:    "namespace",
+		Version:      1,
+		CutoverNanos: 0,
+		MappingRules: map[string]*MappingRuleView{
+			mrIDs[0]: {
+				ID:       mrIDs[0],
+				Name:     "mr",
+				Filter:   "filter",
+				Policies: []policy.Policy{},
+			},
+			mrIDs[1]: {
+				ID:       mrIDs[1],
+				Name:     "mr",
+				Filter:   "filter",
+				Policies: []policy.Policy{},
+			},
+		},
+		RollupRules: map[string]*RollupRuleView{
+			rrIDs[0]: {
+				ID:     rrIDs[0],
+				Name:   "rr",
+				Filter: "filter",
+				Targets: []RollupTargetView{
+					{
+						Name:     "target",
+						Tags:     []string{"tag"},
+						Policies: []policy.Policy{},
+					},
+				},
+			},
+			rrIDs[1]: {
+				ID:     rrIDs[1],
+				Name:   "rr",
+				Filter: "filter",
+				Targets: []RollupTargetView{
+					{
+						Name:     "target",
+						Tags:     []string{"tag"},
+						Policies: []policy.Policy{},
+					},
+				},
+			},
+		},
+	}
+	require.EqualValues(t, expected, actual)
+}
+
+func TestRuleSetSnapshotFailMissingMappingRuleID(t *testing.T) {
+	mappingRules := []MappingRuleJSON{
+		*testMappingRuleJSON("", "mr"),
+		*testMappingRuleJSON("id1", "mr"),
+	}
+	rollupRules := []RollupRuleJSON{
+		*testRollupRuleJSON("id2", "rr", []RollupTargetJSON{*testRollupTargetJSON("target")}),
+		*testRollupRuleJSON("id3", "rr", []RollupTargetJSON{*testRollupTargetJSON("target")}),
+	}
+	fixture := testRuleSetJSON("namespace", mappingRules, rollupRules)
+
+	_, err := fixture.RuleSetSnapshot(DontGenerateID)
+	require.Error(t, err)
+}
+
+func TestRuleSetSnapshotFailMissingRollupRuleID(t *testing.T) {
+	mappingRules := []MappingRuleJSON{
+		*testMappingRuleJSON("id1", "mr"),
+		*testMappingRuleJSON("id2", "mr"),
+	}
+	rollupRules := []RollupRuleJSON{
+		*testRollupRuleJSON("id3", "rr", []RollupTargetJSON{*testRollupTargetJSON("target")}),
+		*testRollupRuleJSON("", "rr", []RollupTargetJSON{*testRollupTargetJSON("target")}),
+	}
+	fixture := testRuleSetJSON("namespace", mappingRules, rollupRules)
+
+	_, err := fixture.RuleSetSnapshot(DontGenerateID)
+	require.Error(t, err)
+}
+
+func TestRuleSetsSort(t *testing.T) {
```

Comment:
Can you rename this to `TestRuleSetJSONsSort`? Please also rename the rest of the test functions so the name of the type in the test function matches the name of the type you are testing.